pkg/agent/controller/controller_suite_test.go

```diff
@@ -1059,28 +1060,22 @@ func (t *testDriver) awaitServiceUnexported(c *cluster) {
 	Expect(epsClient.Delete(context.Background(), "dummy", metav1.DeleteOptions{})).To(Succeed())
 }
 
-func newServiceExportValidCondition(status metav1.ConditionStatus, reason string) *metav1.Condition {
-	return &metav1.Condition{
-		Type:   mcsv1a1.ServiceExportValid,
-		Status: status,
-		Reason: reason,
-	}
+func newServiceExportValidCondition(status metav1.ConditionStatus, reason mcsv1a1.ServiceExportConditionReason) metav1.Condition {
+	return mcsv1a1.NewServiceExportCondition(mcsv1a1.ServiceExportConditionValid, status, reason, "")
 }
 
-func newServiceExportReadyCondition(status metav1.ConditionStatus, reason string) *metav1.Condition {
-	return &metav1.Condition{
-		Type:   constants.ServiceExportReady,
-		Status: status,
-		Reason: reason,
-	}
+func newServiceExportReadyCondition(status metav1.ConditionStatus, reason mcsv1a1.ServiceExportConditionReason) metav1.Condition {
+	return mcsv1a1.NewServiceExportCondition(mcsv1a1.ServiceExportConditionReady, status, reason, "")
 }
 
-func newServiceExportConflictCondition(reason string) *metav1.Condition {
-	return &metav1.Condition{
-		Type:   mcsv1a1.ServiceExportConflict,
-		Status: metav1.ConditionTrue,
-		Reason: reason,
+func newServiceExportConflictCondition(reason ...mcsv1a1.ServiceExportConditionReason) metav1.Condition {
+	c := mcsv1a1.NewServiceExportCondition(mcsv1a1.ServiceExportConditionConflict, metav1.ConditionTrue, reason[0], "")
+
+	for i := 1; i < len(reason); i++ {
+		c.Reason += "," + string(reason[i])
 	}
+
+	return c
```

Comment:
```suggestion
	return mcsv1a1.NewServiceExportCondition(mcsv1a1.ServiceExportConditionConflict, metav1.ConditionTrue, strings.Join(reason, ","), "")
```

to avoid a panic if no reason is given.